internal/contentmanager/cache.go

```diff
@@ -0,0 +1,115 @@
+package contentmanager
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// Cache is a storage mechanism for keeping track of
+// managed content sources. It also exposes methods
+// for retrieving the stored references of the managed
+// content
+type Cache interface {
```

Comment:
Seeing another custom cache implementation sort of sounded alarm-bells for me coming from v0's and all the cache issues there but this seems kind of unfair to compare to that as long as it stays as simple as it is written here.

Reply:
I agree that there is a negative history with needing a custom caching layer. I 1000% agree with you that we need to ensure this caching layer is as simple as possible.

Unfortunately I can't see any other way of achieving what we are looking for. If naming is an issue I'm open to suggestions for naming other than "Cache" :)